test/utils.ts

```diff
@@ -0,0 +1,66 @@
+
+import { Stats, stat, readFile, writeFile } from 'fs';
+import * as glob from 'glob';
+import { ncp } from 'ncp';
+import * as path from 'path';
+import * as pify from 'pify';
+import { ChildProcess, ForkOptions, fork, SpawnOptions, spawn } from 'child_process';
+import * as once from 'once';
+import * as tmp from 'tmp';
+import * as rimraf from 'rimraf';
+
+export const BUILD_DIRECTORY = 'build';
+
+export const globP: (pattern: string) => Promise<string[]> = pify(glob);
+export const ncpP: (src: string, dest: string) => Promise<void> = pify(ncp);
+export const readFileP: (path: string, encoding?: string) => Promise<Buffer|string> = pify(readFile);
+export const writeFileP: (path: string, data: string, encoding?: string) => Promise<void> = pify(writeFile);
+export const statP: (path: string) => Promise<Stats> = pify(stat);
+export const tmpDirP: () => Promise<string> = pify(tmp.dir);
+export const rimrafP: (f: string) => Promise<void> = pify(rimraf);
+
+export function nodule(nodule: string) {
+  return path.relative(BUILD_DIRECTORY, `node_modules/${nodule}`);
+}
+
+export function existsP(path: string): Promise<boolean> {
```

Comment:
I think it might be more helpful to change the name of the function to something more descriptive like expanding the meaning of 'P' as some of the function name P mean 'path' where others mean 'process'.

Reply:
The P in this case refers to Promise.  This is the same code used by the trace agent, but we could use different names if it makes it easier to reason about the code.

Reply:
I prefer the `-P` suffix given that all pify'd functions already exist purely for convenience. Having to type out `Promise` every time in place of `P` is cumbersome.

I remember replicating this pattern from another codebase... not sure where it started.